doc/srv6/srv6_vpn.md

```diff
@@ -0,0 +1,287 @@
+    
+    *   Verify that the SAI object created in the above step are deleted
+        
+5.  Switch SRv6 route from single nexthop to nexthopgroup. Verify the following items
+    
+    *   SAI\_OBJECT\_TYPE\_TUNNEL\_MAP\_ENTRY with a new SAI\_TUNNEL\_MAP\_ENTRY\_ATTR\_PREFIX\_AGG\_ID\_KEY is created
+        
+    *   SAI\_OBJECT\_TYPE\_NEXT\_HOP\_GROUP is created
+        
+    *   SAI\_OBJECT\_TYPE\_NEXT\_HOP\_GROUP\_MEMBER is created
+        
+    *   SAI\_OBJECT\_TYPE\_ROUTE\_ENTRY with a new SAI\_ROUTE\_ENTRY\_ATTR\_PREFIX\_AGG\_ID and a new SAI\_ROUTE\_ENTRY\_ATTR\_NEXT\_HOP\_ID is seted
+        
+6.  Add two SRv6 route with same nexthopgroup. Verify the following items
+    
+    *   Add first one, Verify SAI object in step 3
+        
+    *   Add second one, Verify SAI\_OBJECT\_TYPE\_ROUTE\_ENTRY is created,  with same SAI\_ROUTE\_ENTRY\_ATTR\_PREFIX\_AGG\_ID and SAI\_ROUTE\_ENTRY\_ATTR\_NEXT\_HOP\_ID as first one 
+        
+7.  Remove two SRv6 route one by one. Verify the following items
+    
+    *   Delete first route,  Verify SAI\_OBJECT\_TYPE\_ROUTE\_ENTRY is deleted
+        
+    *   Delete second one, Verify that the SAI object created in the step 6 are deleted
+        
+
+### sbfd
+
+1.  Add and remove ipv6 seamless BFD session (sbfd initiator and sbfd reflector). Verify the asci db  especially SAI\_BFD\_SESSION\_ATTR\_BFD\_ENCAPSULATION\_TYPE and SAI\_BFD\_SESSION\_ATTR\_SRV6\_SIDLIST\_ID and state db. 
+    
+2.  Add and remove multiple ipv6 seamless BFD session . Verify same as last one.
+    
+3.  Add and remove ipv6 seamless BFD echo . Verify the asci db  especially SAI\_BFD\_SESSION\_ATTR\_BFD\_ENCAPSULATION\_TYPE, SAI\_BFD\_SESSION\_ATTR\_ECHO\_ENABLE , SAI\_BFD\_SESSION\_ATTR\_SRV6\_SIDLIST\_ID.
+    
+4.  Add and remove multiple ipv6 seamless BFD echo . Verify same as last one.
+    
+5.  Handle  the notify of bfd hardware state's change.  Verify the state db.
+    
+
+# References
+
+RFC 8986 Segment Routing over IPv6 (SRv6) Network Programmin
+
+RFC 9256 Segment Routing Policy Architecture
+
+RFC 7880 Seamless Bidirectional Forwarding Detection (S-BFD)
+
+RFC 7881Seamless Bidirectional Forwarding Detection (S-BFD) for IPv4, IPv6, and MPLS
+
+# Current SONiC's Design Limitations and workarounds
+
+We don't have an efficient infrastructure to communicate between FRR related processes and SONiC related processes due to current SONiC design limitation. This area needs to be redesigned in the future. Alibaba's business can't wait for the community agreed solution to be ready, therefore we put in the following workaround for an overall solution in the cloud environment. These workarounds would NOT be commited in 202305 release. We will raise PRs and provide diff for referencing purposes only. 
+
+## SRv6 policy download
+
+For a simple workaround, FRR pathd would add, update and delete SRv6 policy related information by writing to APP\_DB directly.
```

Comment:
IMO, we should use fpmsyncd path to program the routes, policy..etc to APP_DB to avoid multiple sources updating the APP_DB for SRv6. let's understand the pain points we have with 2M routes.

Reply:
@venkatmahalingam  there is no question on writing to APP_DB. The question is really how. That's why I want to start a routing working group to figure out what is the best and efficient way to set up this communication channel, bi-directional.

Reply:
OK, thanks!